[ASAN_X] heap-use-after-free in NoProductResolverException raised from alpaka_serial_sync::TestAlpakaGlobalProducerNullES::produce

In CMSSW_15_1_ASAN_X_2025-03-31-2300, ASAN reports heap-use-after-free error:

```
=================================================================
==3552287==ERROR: AddressSanitizer: heap-use-after-free on address 0x602000137430 at pc 0x15500d2a8e37 bp 0x7ffcc212cb90 sp 0x7ffcc212c340
READ of size 1 at 0x602000137430 thread T0
    #0 0x15500d2a8e36 in __interceptor_strncmp ../../../../libsanitizer/sanitizer_common/sanitizer_common_interceptors.inc:497
    #1 0x15500c9fe8c0 in edm::eventsetup::NoProductResolverException::NoProductResolverException(edm::eventsetup::EventSetupRecordKey const&, char const*, char const*, bool) (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/libFWCoreFramework.so+0x5fe8c0)
    #2 0x154fe7f05b8a in edm::eventsetup::EventSetupRecord::noResolverHandle<edm::ESHandle, PortableHostCollection<cms::alpakatest::AlpakaESTestSoALayoutC<128ul, false> >, AlpakaESTestRecordA>(edm::ESGetToken<PortableHostCollection<cms::alpakatest::AlpakaESTestSoALayoutC<128ul, false> >, AlpakaESTestRecordA> const&, bool) const::{lambda()#1}::operator()() const (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/pluginHeterogeneousCoreAlpakaTestPluginsPortableSerialSync.so+0x305b8a)
    #3 0x154fe7f05e14 in edm::FunctorESHandleExceptionFactory<edm::eventsetup::EventSetupRecord::noResolverHandle<edm::ESHandle, PortableHostCollection<cms::alpakatest::AlpakaESTestSoALayoutC<128ul, false> >, AlpakaESTestRecordA>(edm::ESGetToken<PortableHostCollection<cms::alpakatest::AlpakaESTestSoALayoutC<128ul, false> >, AlpakaESTestRecordA> const&, bool) const::{lambda()#1}>::make() const (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/pluginHeterogeneousCoreAlpakaTestPluginsPortableSerialSync.so+0x305e14)
    #4 0x154fe7cb1df7 in edm::ESHandleBase::productStorage() const [clone .isra.0] (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/pluginHeterogeneousCoreAlpakaTestPluginsPortableSerialSync.so+0xb1df7)
    #5 0x154fe7f04eff in alpaka_serial_sync::TestAlpakaGlobalProducerNullES::produce(edm::StreamID, alpaka_serial_sync::device::Event&, alpaka_serial_sync::device::EventSetup const&) const (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/pluginHeterogeneousCoreAlpakaTestPluginsPortableSerialSync.so+0x304eff)
    #6 0x154fe7e7af32 in virtual thunk to alpaka_serial_sync::global::EDProducer<>::produce(edm::StreamID, edm::Event&, edm::EventSetup const&) const (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/pluginHeterogeneousCoreAlpakaTestPluginsPortableSerialSync.so+0x27af32)
    #7 0x15500ce92424 in edm::global::EDProducerBase::doEvent(edm::EventTransitionInfo const&, edm::ActivityRegistry*, edm::ModuleCallingContext const*) (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/libFWCoreFramework.so+0xa92424)
    #8 0x15500ce75318 in edm::WorkerT<edm::global::EDProducerBase>::implDo(edm::EventTransitionInfo const&, edm::ModuleCallingContext const*) (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/libFWCoreFramework.so+0xa75318)
    #9 0x15500ca80b99 in decltype ({parm#1}()) edm::convertException::wrap<edm::Worker::runModule<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::TransitionInfoType const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*)::{lambda()#1}>(edm::Worker::runModule<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::TransitionInfoType const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*)::{lambda()#1}) (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/libFWCoreFramework.so+0x680b99)
    #10 0x15500ca8129b in std::__exception_ptr::exception_ptr edm::Worker::runModuleAfterAsyncPrefetch<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(std::__exception_ptr::exception_ptr, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::TransitionInfoType const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*) (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/libFWCoreFramework.so+0x68129b)
    #11 0x15500ca91dec in edm::Worker::RunModuleTask<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >::execute() (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/libFWCoreFramework.so+0x691dec)
    #12 0x15500d904072 in tbb::detail::d2::function_task<edm::WaitingTaskList::announce()::{lambda()#1}>::execute(tbb::detail::d1::execution_data&) (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/libFWCoreConcurrency.so+0x17072)
    #13 0x15500be2587a in tbb::detail::d1::task* tbb::detail::r1::task_dispatcher::local_wait_for_all<false, tbb::detail::r1::external_waiter>(tbb::detail::d1::task*, tbb::detail::r1::external_waiter&) /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc12/external/tbb/v2022.0.0-10fdc07bd3d228f4c274203a8cd20b5c/tbb-v2022.0.0/src/tbb/task_dispatcher.h:334
    #14 0x15500be2587a in tbb::detail::d1::task* tbb::detail::r1::task_dispatcher::local_wait_for_all<tbb::detail::r1::external_waiter>(tbb::detail::d1::task*, tbb::detail::r1::external_waiter&) /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc12/external/tbb/v2022.0.0-10fdc07bd3d228f4c274203a8cd20b5c/tbb-v2022.0.0/src/tbb/task_dispatcher.h:470
    #15 0x15500be2587a in tbb::detail::r1::task_dispatcher::execute_and_wait(tbb::detail::d1::task*, tbb::detail::d1::wait_context&, tbb::detail::d1::task_group_context&) /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc12/external/tbb/v2022.0.0-10fdc07bd3d228f4c274203a8cd20b5c/tbb-v2022.0.0/src/tbb/task_dispatcher.cpp:168
    #16 0x15500c7ac98c in edm::FinalWaitingTask::wait() (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/libFWCoreFramework.so+0x3ac98c)
    #17 0x15500c744e8c in edm::EventProcessor::processRuns() (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/libFWCoreFramework.so+0x344e8c)
    #18 0x15500c77e14d in edm::EventProcessor::runToCompletion() (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/lib/el8_amd64_gcc12/libFWCoreFramework.so+0x37e14d)
    #19 0x40ceb4 in tbb::detail::d1::task_arena_function<main::{lambda()#1}::operator()() const::{lambda()#1}, void>::operator()() const (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/bin/el8_amd64_gcc12/cmsRun+0x40ceb4)
    #20 0x15500be13f70 in tbb::detail::r1::task_arena_impl::execute(tbb::detail::d1::task_arena_base&, tbb::detail::d1::delegate_base&) /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc12/external/tbb/v2022.0.0-10fdc07bd3d228f4c274203a8cd20b5c/tbb-v2022.0.0/src/tbb/arena.cpp:821
    #21 0x414958 in main::{lambda()#1}::operator()() const (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/bin/el8_amd64_gcc12/cmsRun+0x414958)
    #22 0x4087d6 in main (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/bin/el8_amd64_gcc12/cmsRun+0x4087d6)
    #23 0x15500b0647e4 in __libc_start_main (/lib64/libc.so.6+0x3a7e4)
    #24 0x408b8d in _start (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/bin/el8_amd64_gcc12/cmsRun+0x408b8d)

0x602000137430 is located 0 bytes inside of 16-byte region [0x602000137430,0x602000137440)
freed by thread T0 here:
    #0 0x15500d2b78b8 in __interceptor_free ../../../../libsanitizer/asan/asan_malloc_linux.cpp:52
    #1 0x154fffc87228 in clang::TreeTransform<(anonymous namespace)::TemplateInstantiator>::TransformNestedNameSpecifierLoc(clang::NestedNameSpecifierLoc, clang::QualType, clang::NamedDecl*) (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/external/el8_amd64_gcc12/lib/libCling.so+0x2087228)

previously allocated by thread T0 here:
    #0 0x15500d2b8bdf in __interceptor_malloc ../../../../libsanitizer/asan/asan_malloc_linux.cpp:69
    #1 0x1550017342a3 in clang::NestedNameSpecifierLocBuilder::Extend(clang::ASTContext&, clang::SourceLocation, clang::TypeLoc, clang::SourceLocation) (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02883/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_ASAN_X_2025-03-31-2300/external/el8_amd64_gcc12/lib/libCling.so+0x3b342a3)
```

Full log: [link](https://cmssdt.cern.ch/SDT/cgi-bin/logreader/el8_amd64_gcc12/CMSSW_15_1_ASAN_X_2025-03-31-2300/unitTestLogs/HeterogeneousCore/AlpakaTest#/35)

It is hard to tell. When the crash occured, scram was building 3 tests and running one

```
$ cat unitTests1.log | grep -v Pass | grep -v Skip
>> Local Products Rules ..... started
>> Local Products Rules ..... done
Creating test log file logs/el8_amd64_gcc12/testing.log
gmake: *** [config/SCRAM/GMake/Makefile.rules:1886: unittests_test_MC_23_crosscheck] Terminated
gmake: *** [config/SCRAM/GMake/Makefile.rules:1886: unittests_PrimaryVertex] Terminated
gmake: *** [config/SCRAM/GMake/Makefile.rules:1886: unittests_testTrackAnalysis] Terminated
/bin/sh: line 1: 2935130 Terminated              ( ( /usr/bin/mkdir -p /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/unit_tests/test_PixelBaryCentreTool && export CMSSW_SEARCH_PATH=/data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/unit_tests/test_PixelBaryCentreTool:$CMSSW_SEARCH_PATH && cd /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/unit_tests/test_PixelBaryCentreTool && LD_PRELOAD=/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02884/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib64/libasan.so timeout 9000 sh -c '/data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/src/Alignment/TrackerAlignment/test/pixelPositions.sh' && /usr/bin/echo "" && /usr/bin/echo "---> test test_PixelBaryCentreTool succeeded" ) >> tmp/el8_amd64_gcc12/src/Alignment/TrackerAlignment/test/test_PixelBaryCentreTool/testing.log 2>&1 || ( /usr/bin/echo "" && /usr/bin/echo "---> test test_PixelBaryCentreTool had ERRORS" ) >> tmp/el8_amd64_gcc12/src/Alignment/TrackerAlignment/test/test_PixelBaryCentreTool/testing.log )
/bin/sh: line 1: 2935189 Terminated              ( ( /usr/bin/mkdir -p /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/unit_tests/test_ProduceSystematicMisalignment && export CMSSW_SEARCH_PATH=/data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/unit_tests/test_ProduceSystematicMisalignment:$CMSSW_SEARCH_PATH && cd /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/unit_tests/test_ProduceSystematicMisalignment && LD_PRELOAD=/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02884/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib64/libasan.so timeout 9000 sh -c '/data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/src/Alignment/TrackerAlignment/test/testProduceAllMisalignments.sh' && /usr/bin/echo "" && /usr/bin/echo "---> test test_ProduceSystematicMisalignment succeeded" ) >> tmp/el8_amd64_gcc12/src/Alignment/TrackerAlignment/test/test_ProduceSystematicMisalignment/testing.log 2>&1 || ( /usr/bin/echo "" && /usr/bin/echo "---> test test_ProduceSystematicMisalignment had ERRORS" ) >> tmp/el8_amd64_gcc12/src/Alignment/TrackerAlignment/test/test_ProduceSystematicMisalignment/testing.log )
/bin/sh: line 1: 2902980 Terminated              ( /usr/bin/mkdir -p /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/unit_tests/Miscellanea && export CMSSW_SEARCH_PATH=/data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/unit_tests/Miscellanea:$CMSSW_SEARCH_PATH && cd /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/unit_tests/Miscellanea && LD_PRELOAD=/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02884/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib64/libasan.so timeout 9000 sh -c '/data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/src/Alignment/OfflineValidation/test/testingScripts/test_unitMiscellanea.sh' && /usr/bin/echo "" && /usr/bin/echo "---> test Miscellanea succeeded" ) >> tmp/el8_amd64_gcc12/src/Alignment/OfflineValidation/test/Miscellanea/testing.log 2>&1
/bin/sh: line 1: 2902979 Terminated              ( ( /usr/bin/mkdir -p /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/unit_tests/Miscellanea && export CMSSW_SEARCH_PATH=/data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/unit_tests/Miscellanea:$CMSSW_SEARCH_PATH && cd /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/unit_tests/Miscellanea && LD_PRELOAD=/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02884/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib64/libasan.so timeout 9000 sh -c '/data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/src/Alignment/OfflineValidation/test/testingScripts/test_unitMiscellanea.sh' && /usr/bin/echo "" && /usr/bin/echo "---> test Miscellanea succeeded" ) >> tmp/el8_amd64_gcc12/src/Alignment/OfflineValidation/test/Miscellanea/testing.log 2>&1 || ( /usr/bin/echo "" && /usr/bin/echo "---> test Miscellanea had ERRORS" ) >> tmp/el8_amd64_gcc12/src/Alignment/OfflineValidation/test/Miscellanea/testing.log )
gmake: *** [config/SCRAM/GMake/Makefile.rules:1886: unittests_Miscellanea] Terminated
gmake: *** [config/SCRAM/GMake/Makefile.rules:1886: unittests_SubmitPVsplit] Terminated
gmake: *** [config/SCRAM/GMake/Makefile.rules:1886: unittests_test_PixelBaryCentreTool] Terminated
gmake: *** [config/SCRAM/GMake/Makefile.rules:1886: unittests_test_ProduceSystematicMisalignment] Terminated
echo: write error: Broken pipe
echo: write error: Broken pipe
echo: write error: Broken pipe
```

Last lines of Zmumuall test log:
```
Building model with signal and backgound
Warning in <ROOT::Math::Fitter::CalculateHessErrors>: Error when calculating Hessian
Info in <TCanvas::Print>: pdf file /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/tmp/UnitTest/test_yaml/Zmumu/single/unitTestZmumuMCreal/testSingleZMM/1/fitResultPlot/EtaMinus_0_oldtool.pdf has been created
Info in <TCanvas::SaveAs>: ROOT file /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/tmp/UnitTest/test_yaml/Zmumu/single/unitTestZmumuMCreal/testSingleZMM/1/fitResultPlot/EtaMinus_0_oldtool.root has been created
void Fitting_GetMassmeanVSvariables(TString, TString) th1d_variables_meanmass[i]->GetNbinsX()=12
void Fitting_GetMassmeanVSvariables(TString, TString) th2d_mass_variables[i]->GetNbinsY()=12
Building model with signal and backgound
/bin/sh: line 1: 2934678 Terminated              LD_PRELOAD=/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02884/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib64/libasan.so timeout 9000 sh -c '/data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_15_1_ASAN_X_2025-04-07-2300/src/Alignment/OfflineValidation/test/testingScripts/test_unitZmumu.sh'
TestTime:5654
^^^^ End Test Zmumuall ^^^^
```

The tests passes locally, so I don't think it's the reason it crashes.

--- a/CondCore/Utilities/test/test_uploadConditions.sh
+++ b/CondCore/Utilities/test/test_uploadConditions.sh
@@ -32,7 +32,7 @@ cat <<EOF >> BasicPayload_v0_ref.txt
 }
 EOF
 
-echo "Content of the directory is:" `ls -lh . | grep db`
+#echo "Content of the directory is:" `ls -lh . | grep db`
 echo -ne '\n\n'
 
 if test -f "BasicPayload_v0.txt"; then

--- a/CondCore/Utilities/test/test_uploadConditions.sh
+++ b/CondCore/Utilities/test/test_uploadConditions.sh
@@ -32,7 +32,7 @@ cat <<EOF >> BasicPayload_v0_ref.txt
 }
 EOF
 
-#echo "Content of the directory is:" `ls -lh . | grep db`
+echo "Content of the directory is:" `ls . | grep db`
 echo -ne '\n\n'
 
 if test -f "BasicPayload_v0.txt"; then

